docs/en/install-upgrade/installing-stack-demo-secure.asciidoc

```diff
@@ -0,0 +1,894 @@
+
+In a production environment you would typically use the CA certificate from your own organization, along with the certificate files generated for the hosts where the {stack} is being installed. For demonstration purposes, we'll use the Elastic certificate utility to configure a self-signed CA certificate.
+
+. On the first node in your {es} cluster, stop the {es} service:
++
+["source","shell"]
+----
+sudo systemctl stop elasticsearch.service
+----
+
+. Generate a CA certificate using the provided certificate utility, `elasticsearch-certutil`. Note that the location of the utility depends on the installation method you used to install {es}. Refer to {ref}/certutil.html[elasticsearch-certutil] for the command details and to {ref}/update-node-certs-different.html[Update security certificates with a different CA] for details about the procedure as a whole.
++
+Run the following command. When prompted, specify a unique name for the output file, such as `elastic-stack-ca-new.zip`: 
++
+["source","shell"]
+----
+sudo /usr/share/elasticsearch/bin/elasticsearch-certutil ca -pem
+----
+
+. Move the output file to the `/etc/elasticsearch/certs` directory. This directory is created automatically when you install {es}.
++
+["source","shell"]
+----
+sudo mv /usr/share/elasticsearch/elastic-stack-ca-new.zip /etc/elasticsearch/certs/
+----
+
+. Change to the certificates directory and unzip the file:
++
+["source","shell"]
+----
+sudo cd /etc/elasticsearch/certs
+sudo unzip elastic-stack-ca-new.zip
+----
+
+. View the files that were unpacked into a new `ca` directory:
++
+["source","shell"]
+----
+sudo ls ca/
+----
++
+`ca.crt`:: The generated certificate (or you can substitute this with your own certificate, signed by your organizations's certificate authority)
+`ca.key`:: The certificate authority's private key
+
++
+These steps to generate new self-signed CA certificates need to be done only on the first {es} node. The other {es} nodes will use the same `ca.crt` and `ca.key` files.
+
+. From the `/ca` directory, import the newly created CA certificate into the {es} truststore. This step ensures that your cluster trusts the new CA certificate.
++
+NOTE: On a new installation a new keystore and truststore are created automatically. If you're running these steps on an existing {es} installation and you know the password to the keystore and the truststore, follow the instructions in {ref}/update-node-certs-different.html[Update security certificates with a different CA] to import the CA certificate.
++
+Run the `keytool` command as shown, replacing `<password>` with a unique password for the truststore, and store the password securely:
++
+["source","shell"]
+----
+sudo /usr/share/elasticsearch/jdk/bin/keytool -importcert -trustcacerts -noprompt -keystore elastic-stack-ca.p12 -storepass <password> -alias new-ca -file ca.crt
+----
+
+. Ensure that the new key was added to the keystore:
++
+["source","shell"]
+----
+keytool -keystore elastic-stack-ca.p12 -list
```

Comment:
I think we should put a note here that says we provide the keytool utility with the elasticsearch installation at: `/usr/share/elasticsearch/jdk/bin/keytool`